vault/diagnose/raft_checks.go

```diff
@@ -0,0 +1,102 @@
+package diagnose
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/vault/physical/raft"
+)
+
+// RaftFilePermsChecks inputs the raft file path and outputs:
+// 1. A bool, which dictates whether Owner has rw permissions
+// 2. A list of errors
+func RaftFilePermsChecks(ctx context.Context, path string) {
+	owner := "owner"
+	group := "group"
+	other := "other"
+	var overPermissions []string
+
+	// Note: Stat does not return information about the symlink itself, in the case where we are dealing with one.
+	info, err := os.Stat(path)
+	if err != nil {
+		SpotError(ctx, "raft file permission checks", fmt.Errorf("error computing file permissions: %w", err))
+	}
+
+	mode := info.Mode()
+
+	canRead := false
+	canWrite := false
+	for i := 1; i < 4; i++ {
+		perm := string(mode.String()[i])
```

Comment:
You should be able to test these using bitwise and.  For example, to test for o:rw, mode & 0700 != 0.  Might be a touch cleaner than string comparison.

Reply:
On the other hand, you're not having to re-synthesize the perm strings, so maybe it's a wash.  Something about this just feels a touch off...

Reply:
Yeah, I did find the bitwise way to do this as well, but I wondered if this would be more human-readable. Either way is okay with me though -- I'll change this to use bitwise operations for now, and we can see if that looks cleaner. If not, I can always revert.